workflow/scripts/tmb.py

```diff
@@ -59,107 +60,118 @@
+        AF_index = 0
+        Caller_index = 0
+        i = 0
+        for info in INFO_list:
+            if info[:3] == "AF=":
+                AF_index = i
+            if info[:8] == "CALLERS=":
+                Caller_index = i
+            i += 1
+        AF = float(INFO_list[AF_index][3:])
+        Callers = INFO_list[Caller_index]
+        nr_Callers = len(Callers.split(","))
+        if nr_Callers < 2:
+            continue
+        VEP_INFO = INFO.split("CSQ=")[1].split("|")
+        Variant_type = VEP_INFO[vep_dict["Consequence"]].split("&")
+        db1000G = VEP_INFO[vep_dict["AF"]]
+        if db1000G == "":
+            db1000G = 0
+        else:
+            db1000G = float(db1000G)
+        GnomAD = VEP_INFO[vep_dict["MAX_AF"]]
+        if GnomAD == "":
+            GnomAD = 0
+        else:
+            GnomAD = float(GnomAD)
+        FORMAT = lline[8].split(":")
+        AD_index = 0
+        DP_index = 0
+        VD_index = 0
+        i = 0
+        for f in FORMAT:
+            if f == "AD":
+                AD_index = i
+            elif f == "DP":
+                DP_index = i
+            elif f == "VD":
+                VD_index = i
+            i += 1
+        DATA = lline[9].split(":")
+        AD = DATA[AD_index].split(",")
+        if len(AD) == 2:
+            VD = int(AD[1])
+        else:
+            VD = int(DATA[VD_index])
+        DP = int(DATA[DP_index])
 
-    # TMB
-    if (filter.find("PASS") != -1 and DP > 200 and VD > 10 and AF >= 0.02 and AF <= 0.45 and
-            GnomAD <= 0.0001 and db1000G <= 0.0001 and Observations < filter_nr_observations and INFO.find("Complex") == -1):
+        # Artifact observations
+        Observations = 0
         if len(ref) == 1 and len(alt) == 1:
-            panel_median = 1000
-            panel_sd = 1000
-            run_median = 1000
-            pos_sd = 1000
-            key2 = key[3:]
-            if key2 in gvcf_panel_dict:
-                panel_median = gvcf_panel_dict[key2][0]
-                panel_sd = gvcf_panel_dict[key2][1]
-            if key2 in gvcf_run_dict:
-                run_median = gvcf_run_dict[key2]
-            if panel_sd > 0.0:
-                pos_sd = (AF - panel_median) / panel_sd
-            if pos_sd > 5.0:
-                if ("missense_variant" in Variant_type or
-                        "stop_gained" in Variant_type or
-                        "stop_lost" in Variant_type):
-                    nr_nsSNV_TMB += 1
-                    TMB_nsSNV.append([line, panel_median, panel_sd, run_median, AF, pos_sd])
-                elif "synonymous_variant" in Variant_type:
-                    nr_sSNV_TMB += 1
-                    TMB_sSNV.append([line, panel_median, panel_sd, run_median, AF, pos_sd])
+            if key in FFPE_SNV_artifacts:
+                Observations = FFPE_SNV_artifacts[key]
+        else:
+            continue
+
+        # TMB
+        if (DP > 200 and VD > 10 and AF >= 0.02 and AF <= 0.45 and
```

Comment:
Shouldn't this parameters be configurable?

Reply:
done